frontend/projects/upgrade/src/app/features/dashboard/segments/components/segments-list/segments-list.component.ts

```diff
@@ -69,13 +79,45 @@ export class SegmentsListComponent implements OnInit, OnDestroy, AfterViewInit {
     this.permissions$ = this.authService.userPermissions$;
     this.allSegmentsSub = this.segmentsService.allSegments$.subscribe((segments) => {
       segments = segments.map((segment) => ({ ...segment, status: segment.status || SEGMENT_STATUS.UNUSED }));
-      this.allSegments = new CustomMatTableSource();
+      this.allSegments = new MatTableDataSource();
       this.allSegments.data = [...segments];
       this.allSegments.sort = this.sort;
       this.cdr.detectChanges();
+      this.applyFilter(this.searchValue);
     });
   }
 
+  filterExperimentPredicate(type: EXPERIMENT_SEARCH_KEY) {
```

Comment:
Change this to `filterSegmentPredicate()`

Reply:
done